modules/ROOT/pages/6.4.1-release-notes.adoc

```diff
@@ -0,0 +1,1334 @@
+
+For information on the **PowerPaste** plugin, see: xref:introduction-to-powerpaste.adoc[PowerPaste].
+
+
+=== Spell Checker Pro 3.2.1
+
+The {productname} 6.4.1 release includes an accompanying release of the **Spell Checker Pro** premium plugin.
+
+**Spell Checker Pro** 3.2.1 includes the following fix.
+
+==== Spell Checker Pro plugin over-loaded the spell checking service
+//#TINY-9391
+
+Previously, the Spell Checker Pro plugin would request each word in the content, and repeatedly request these words every time the spellchecker was activated. This resulted in duplicate words being requested during regular use of the plugin.
+
+With this update, Spell Checker Pro now only sends requests to the spelling service for each unique word and caches these results for later retrieval. This greatly reduces the amount and size of requests sent by the plugin during regular use, improving the reliability of the service and the plugin.
+
+For information on the **Spell Checker Pro** plugin, see: xref:introduction-to-tiny-spellchecker.adoc[Introduction to Spell Checker Pro].
+
+
+[[accompanying-premium-plugin-end-of-life-announcement]]
+== Accompanying Premium Plugin end-of-life announcement
+
+The following premium plugin has been announced as reaching its end-of-life:
+
+=== Real-Time Collaboration
+
+TinyMCE’s xref:rtc-introduction.adoc[Real-time Collaboration (RTC)] plugin will be retired and deactivated on December 31, 2023, and is no longer available for purchase.
+
+
+[[accompanying-premium-skins-and-icon-packs-changes]]
+== Accompanying Premium Skins and Icon Packs changes
+
+The {productname} 6.4.1 release includes an accompanying release of the **Premium Skins and Icon Packs**.
+
+=== Premium Skins and Icon Packs
+
+The **Premium Skins and Icon Packs** release includes the following updates:
+
+The **Premium Skins and Icon Packs** were rebuilt to pull in the changes also incorporated into the default {productname} 6.4.1 skin, Oxide.
+
+For information on using premium skins and icon packs, see: xref:premium-skins-and-icons.adoc[Premium Skins and Icon Packs].
+
+
+[[improvements]]
+== Improvements
+
+{productname} 6.4.1 also includes the following improvements:
+
+=== Direct invalid child text nodes of list elements will be wrapped in list item elements.
+//#TINY-4818
+
+Invalid child nodes were throwing a `Uncaught TypeError:` when `<ol>` tags directly included a text node. This `TypeError` would not allow the user to `Tab` a text node while nested within `<ol>/<li>` HTML tags.
+
+Improvements were made to add a content filter for lists within {productname} 6.4.1. This improvement meant that any _invalid child nodes_ nested within `<li>` elements would be wrapped with this filter to avoid the `Uncaught TypeError:`.
+
+In {productname} 6.4.1, direct _invalid child text nodes_ of list elements are now wrapped in list item elements, which allow the user to now `Tab` nested text nodes that are within `<ol>/<li>` tags.
```

Comment:
```suggestion
In previous versions of {productname}, when pressing `Tab` while the cursor is placed next to a `<li>` element, a console error would be un expectantly be thrown.

Improvements were made that added a content filter for lists within {productname} 6.4. This improvement meant that any _invalid child nodes_ nested within `<li>` elements would be wrapped with this filter to avoid the `Uncaught TypeError:`.

In {productname} 6.4, pressing `Tab` while the caret is nested within `<li>` tags, no longer throws a error to the console.
```

Reply:
@TheSpyder Please see suggested changes

Reply:
Sorry, but no. The bug was not the cursor position alone, it was the cursor position _when the document contained unexpected HTML_. That part is critical to keep.

> a console error would be un expectantly be thrown.

typo here

I also think there are too many commas but that's not a topic for holding up the merge

Reply:
```suggestion
In previous versions of {productname}, pressing `Tab` while the caret is in a `<ul>` or `<ol>` list containing direct child text nodes would unexpectedly cause an `Uncaught TypeError` console error to be thrown.

In the following example,

[source,html]
----
<ul>
  item1
  <li>item2</li>
</ul>
----

if the caret is placed anywhere on either `item1` or `item2`, pressing `Tab` would throw an `Uncaught TypeError` to the console.

In {productname} 6.4.1, a content filter for `<ul>` and `<ol>` lists was added to wrap any _direct child text nodes_ within an `<li>`. This transforms them into standard `<li>` child elements, preventing the `Uncaught TypeError` from being thrown.
```

Reply:
@kemister85 Your change still contains a few inaccuracies regarding the scopes of the bug and fix. Please use my suggestion.

Reply:
I committed your suggesting this morning? Let me check

Reply:
Apologies, Github had shown me the wrong view.